### ISSUE TITLE
create TypeScript bindings

### ISSUE THREAD

**Issue by simbo1905** [simbo1905/thinbus-srp-npm#1]
As requested by https://bitbucket.org/madmarkus/ over on the Java version on bitbucket.

**Comment by simbo1905** [comment]
I am probably going to address this as a rewrite in #3

**Comment by simbo1905** [comment]
Well I have slightly changed my mind on this one.

The public API is hex encoded BigIntegers ready to go into HTTP or coming out of it. I could supply binding and that might prevent someone passing something else but it's hard to imaging a subtle bug here. Also, help me out here as I did don't know TS, but saying its a string isn't really going to stop people passing the wrong string. If TS has a type that is a hex string I can validate me can see that being helpful. 

As I am not a TS dev I don't know how annoying not having bindings is. I am assuming that TS is fully interoperable with JS such that it's not a hassle. Yet TS folks want bindings to avoid bugs. In this case it had to get it wrong but if you do a API that is all string isn't going to keep many bugs away. 

Purists will say ”well let me pass BigIntger” and sure if there is a well behaved BigIntrget ts lib then doing a wrapped would be great. Yet as I am not a TS dev folks might understand why I might not be wanting to invest the effort for marginal gain. 

Now some folks might say ”well I will use a pure TS library”. That's your prerogative. Yet I would recommend against using a security library that isn't battle-tested. This repo is tested against the java version. The Java version was tested against the PHP version and uses a stable Java SRP lib. That was reviewed against SRP in a well-reviewed Java crypto implementation. It was also reviewed against the original SRP implementation. I also checked the BigInteger code against the one written by the SRP inventor. For that reason I will delete any comment if anyone spamming this thread to advertise their TS library: I songs have time to cross-check anyone else version to mine and we don't need lots of security libraries with possible lots if bugs. 

In conclusion, this repo is stable and to the best of my knowledge correct and aims to help the most people which is why its plain old JavaScript. If a TS dev wants to supply bindings to aid other TS devs then I would really welcome a contribution. Yet I am likely to not invest the time myself as I can invest it in making demo apps to promote the use of SRP to help raise awareness and help more sites and apps become secure.
